### PR TITLE
Ap/ibc v2 to v4

### PR DESCRIPTION
**Pull request by snobbee**
- IBC go migration from v2 to v4
- and Cosmos SDK upgrade from v0.45.9 to v0.45.16

### CODE REVIEW

File: x/dispensation/types/types.go [hunk 1]
```diff
@@ -6,20 +6,20 @@ package types
...
-		Height: height,
+		Height:        height,
 	}
 }
 
 func NewQueryRecordsByDistributionNameResponse(distributionRecords DistributionRecords, height int64) QueryRecordsByDistributionNameResponse {
 	return QueryRecordsByDistributionNameResponse{
 		DistributionRecords: &distributionRecords,
-		Height: height,
+		Height:              height,
```
Comment (pgoos) [inline x/dispensation/types/types.go:16]: unnecessary indent

File: x/dispensation/types/types.go [hunk 2]
```diff
@@ -6,20 +6,20 @@ package types
...
 	}
 }
 
 func NewQueryRecordsByRecipientAddrResponse(distributionRecords DistributionRecords, height int64) QueryRecordsByRecipientAddrResponse {
 	return QueryRecordsByRecipientAddrResponse{
 		DistributionRecords: &distributionRecords,
-		Height: height,
+		Height:              height,
```
Comment (pgoos) [inline x/dispensation/types/types.go:23]: unnecessary indent

**Review by pgoos — APPROVED** [review]

File: x/dispensation/types/types.go (hunk 1 above)
Comment (snobbee) [reply x/dispensation/types/types.go:16]: @pgoos this is an auto formatting mechanism that is part of the analysis tool in the golang toolset.

File: x/dispensation/types/types.go (hunk 2 above)
Comment (snobbee) [reply x/dispensation/types/types.go:23]: same here

**Pull request merged**